### PR TITLE
capturd exception conserved [API-1003]

### PR DESCRIPTION
**Pull request by emreyigit**
Captured exception is conserved in Hazelcast exception, and related warnings is added to description.

### CODE REVIEW

File: src/Hazelcast.Net/Clustering/MemberConnection.cs [hunk 1]
```diff
@@ -375,6 +375,7 @@ public async Task<ClientMessage> SendAsync(ClientMessage message)
         /// </summary>
         /// <param name="invocation">The invocation.</param>
         /// <param name="cancellationToken">A cancellation token.</param>
+        /// <remarks>After first connection is established, <b>TargetDisconnectedException</b> can be throwed. If connected member registers to cluster with different IP address apart from configured one. In this case, connection will be swicthed, and exception will be throwed.</remarks>
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net/Clustering/MemberConnection.cs:378]:
```suggestion
        /// <remarks>After first connection is established, <b>TargetDisconnectedException</b> can be throwed. If
        // the connected member reports an IP address different from the configured one. In this case, connection 
        // will be switched, and an exception will be thrown.</remarks>
```

**Review by zpqrtbnk — CHANGES_REQUESTED** [review]
One small change proposal and then I am OK.

**Review by zpqrtbnk — APPROVED** [review]
All good.

File: src/Hazelcast.Net.Tests/Sql/SqlQueryResultTests.cs [hunk 2]
```diff
@@ -173,5 +174,25 @@ await foreach (var row in result)
...
+        public async Task SqlErrorHasSuggestion()
+        {
+            string dummyMapName = "testingMap";
+            var map = await Client.GetMapAsync<int, string>(dummyMapName);
+            await map.PutAsync(0, "some value");
+
+            try
+            {
+                //query the map without creating mapping to get exception with suggestion in it.
+                var result = await Client.Sql.ExecuteQueryAsync($"SELECT * FROM {dummyMapName}");
```
[the remark this answers is not in the document]
Comment (emreyigit) [reply src/Hazelcast.Net.Tests/Sql/SqlQueryResultTests.cs:189]: I'm afraid I didn't get it. As long as I don't create mapping on server, I know that it throws `HazelcastSqlException`. If there is another type of exception occurring, that can fail the testing process in anyway.

[the remark this answers is not in the document]
Comment (zpqrtbnk) [reply src/Hazelcast.Net.Tests/Sql/SqlQueryResultTests.cs:189]: Think what Ihsan means is: you expect an exception, so you should test and ensure that an exception is indeed thrown. Here, if no exception is thrown, the test will pass, won't it? We have that:
```
var e = await AssertEx.ThrowsAsync(async () => result = await Client.Sql.ExecuteWhatever());
Assert.IsNotEmpty(e.Message);
```
If the code passes without throwing, `AssertEx` will throw a test exception. See?

[the remark this answers is not in the document]
Comment (emreyigit) [reply src/Hazelcast.Net.Tests/Sql/SqlQueryResultTests.cs]: @ihsandemir, test is improved.

File: src/Hazelcast.Net/Clustering/MemberConnection.cs [hunk 3]
```diff
@@ -375,6 +375,9 @@ public async Task<ClientMessage> SendAsync(ClientMessage message)
         /// </summary>
         /// <param name="invocation">The invocation.</param>
         /// <param name="cancellationToken">A cancellation token.</param>
+        /// <remarks>After first connection is established, <b>TargetDisconnectedException</b> can be throwed. If
```
Comment (ihsandemir) [inline src/Hazelcast.Net/Clustering/MemberConnection.cs:378]:
```suggestion
        /// <remarks>After first connection is established, <b>TargetDisconnectedException</b> may be thrown. If
```

**Review by ihsandemir — APPROVED** [review]

**Pull request merged**